openedx/core/djangoapps/discussions/tests/test_tasks.py

```diff
@@ -105,7 +105,7 @@ def test_default(self):
         assert config_data.course_key == self.course.id
         assert config_data.enable_graded_units is False
         assert config_data.unit_level_visibility is False
-        assert config_data.provider_type is None
+        assert config_data.provider_type is not None
```

Comment:
Initially this signal was avoiding setting a provider type if not already set, but since the provider is set in the database then flows to the course structure, and this signals approach is about changes to the course structure that flow to the database, there shouldn't be any conflict in where this value originates.